### PR TITLE
feat: support `LargeList` in `array_dims`

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Parts #8185

### CODE REVIEW

**Review by jayzhan211 — APPROVED** [review]
LGTM!

File: datafusion/physical-expr/src/array_expressions.rs [hunk 1]
```diff
@@ -1925,12 +1925,29 @@ pub fn array_length(args: &[ArrayRef]) -> Result<ArrayRef> {
 
 /// Array_dims SQL function
 pub fn array_dims(args: &[ArrayRef]) -> Result<ArrayRef> {
-    let list_array = as_list_array(&args[0])?;
+    let data = match args[0].data_type() {
```
Comment (comphead) [inline datafusion/physical-expr/src/array_expressions.rs:1928]: please check args not empty, especially if its pub function

File: datafusion/sqllogictest/test_files/array.slt [hunk 2]
```diff
@@ -2830,6 +2839,11 @@ select array_dims(make_array(1, 2, 3)), array_dims(make_array([1, 2], [3, 4])),
 ----
 [3] [2, 2] [1, 1, 1, 2, 1]
 
+query ???
+select array_dims(arrow_cast(make_array(1, 2, 3), 'LargeList(Int64)')), array_dims(arrow_cast(make_array([1, 2], [3, 4]), 'LargeList(List(Int64))')), array_dims(arrow_cast(make_array([[[[1], [2]]]]), 'LargeList(List(List(List(List(Int64)))))'));
+----
+[3] [2, 2] [1, 1, 1, 2, 1]
```
Comment (comphead) [inline datafusion/sqllogictest/test_files/array.slt:2845]: 👍

**Review by comphead — COMMENTED** [review]
looks great @Weijun-H please address a minor comment

File: datafusion/physical-expr/src/array_expressions.rs (hunk 1 above)
Comment (Weijun-H) [reply datafusion/physical-expr/src/array_expressions.rs:1932]: I will address this later in the pull request as other functions may also lack this check.

**Review by comphead — APPROVED** [review]
lgtm thanks @Weijun-H

**Pull request merged**